--- /home/jtratt/source/netHack-NetHack-3.6.6_Released/win/curses/cursmesg.c	2020-03-08 12:29:31.000000000 -0500
+++ /home/jtratt/source/voice_Nethack_366/win/curses/cursmesg.c	2025-01-11 14:08:43.899220377 -0600
@@ -8,6 +8,8 @@
 #include "wincurs.h"
 #include "cursmesg.h"
 #include <ctype.h>
+#include <stdlib.h>
+#include <unistd.h>
 
 /* defined in sys/<foo>/<foo>tty.c or cursmain.c as last resort;
    set up by curses_init_nhwindows() */
@@ -137,7 +139,7 @@
 
     bold = (height > 1 && !last_messages);
     if (bold)
-        curses_toggle_color_attr(win, NONE, A_BOLD, ON);
+        curses_toggle_color_attr(win, CLR_GREEN, A_BOLD, ON);
 
     /* will this message fit as-is or do we need to split it? */
     if (mx == border_space && message_length > width - 2) {
@@ -150,16 +152,216 @@
             ++mx;
         free(tmpstr);
         if (bold)
-            curses_toggle_color_attr(win, NONE, A_BOLD, OFF);
+            curses_toggle_color_attr(win, CLR_GREEN, A_BOLD, OFF);
         tmpstr = curses_str_remainder(message, (width - 2), 1);
         curses_message_win_puts(tmpstr, TRUE);
         free(tmpstr);
     } else {
         mvwprintw(win, my, mx, "%s", message), mx += message_length;
         if (bold)
-            curses_toggle_color_attr(win, NONE, A_BOLD, OFF);
+            curses_toggle_color_attr(win, CLR_GREEN, A_BOLD, OFF);
     }
     wrefresh(win);
+
+    int status;
+    char txt2sp[255] = "";
+    char tosay[255] = "";
+    char sayit[255] = "";
+    char *match;
+    char *pch;
+    int ii,jj,kk=0;
+    //char str[] = "HELLO'All";
+    char rep[255]={0};
+    //match = "Welcome";
+
+        //memcpy(txt2sp,"/usr/bin/pico2wave -w=/tmp/test.wav ", 255);
+        //memcpy(txt2sp,"/usr/bin/espeak -a 90 -v en-us -g 1 -k 10 -s 250 -z ", 255);
+        /* create batch file containing:'/usr/bin/espeak -a 90 -v en-us -g 1 -k 10 -s 250 -z "$1"' */
+        memcpy(txt2sp,"~/bin/myespeak ", 255); /* requires trialing space */
+        txt2sp[255] = '\0';
+        memcpy(tosay, message, 255);
+        tosay[255] = '\0';
+
+        /* eliminate troublesome characters passed to sh */
+        kk=strlen(tosay);
+    for(ii=jj=0; ii< kk; ii++)
+    {
+      if(tosay[ii]=='\'') /* ' */
+      {
+          //pline("\nB:%s",rep);
+        if (ii >= kk) /* skip if <kk else insert NULL terminator */
+            rep[jj]='\0';
+          //pline("\nA:%s",rep);
+        continue;
+    } else if(tosay[ii]=='\"') /* " */
+      {
+        if (ii >= kk) /* skip if <kk else insert NULL terminator */
+            rep[jj]='\0';
+        continue;
+    } else if(tosay[ii]=='`') /* ` */
+      {
+        if (ii >= kk) /* skip if <kk else insert NULL terminator */
+            rep[jj]='\0';
+        continue;
+    }  else if(tosay[ii]=='|') /* | */
+      {
+        rep[jj]='_';
+        rep[++jj]=tosay[ii];
+        continue;
+    }  else if(tosay[ii]=='>') /* > */
+      {
+        rep[jj]='_';
+        rep[++jj]=tosay[ii];
+        continue;
+    } else if(tosay[ii]=='<') /* < */
+      {
+        rep[jj]='_';
+        rep[++jj]=tosay[ii];
+        continue;
+    } else if(tosay[ii]=='(') /* ( */
+      {
+        rep[jj]='_';
+        rep[++jj]=tosay[ii];
+        continue;
+    } else if(tosay[ii]==')') /* ) */
+      {
+        rep[jj]='_';
+        rep[++jj]=tosay[ii];
+        continue;
+    } else if(tosay[ii]==';') /* ; */
+      {
+        rep[jj]='_';
+        rep[++jj]=tosay[ii];
+        continue;
+    }
+
+
+    rep[jj++]=tosay[ii];
+  }
+
+        memcpy(tosay, tosay, 255);
+        tosay[255] = '\0';
+
+        memcpy(sayit, txt2sp, 17);
+        sayit[17] = '\0';
+        strcat(sayit, "\"");
+        strcat(sayit, rep);
+        strcat(sayit, "\"");
+//pline("tosay: %s", sayit);
+    status = 0;
+    match = "welcome"; pch = strstr(message, match); if (pch) status++;
+    match = "What do you want to"; pch = strstr(message, match); if (pch) status++;
+    //match = "What do you want to rub on the stone"; pch = strstr(message, match); if (pch) status++;
+    //match = "What do you want to drop"; pch = strstr(message, match); if (pch) status++;
+    //match = "What do you want to use or apply"; pch = strstr(message, match); if (pch) status++;
+    match = "Choice: h"; pch = strstr(message, match); if (pch) status++;
+    match = "Choice: l"; pch = strstr(message, match); if (pch) status++;
+    match = "Choice: p"; pch = strstr(message, match); if (pch) status++;
+    match = "Choice: b"; pch = strstr(message, match); if (pch) status++;
+    match = "Choice: g"; pch = strstr(message, match); if (pch) status++;
+    match = "Choice: o"; pch = strstr(message, match); if (pch) status++;
+    match = "Choice: s"; pch = strstr(message, match); if (pch) status++;
+    match = "Choice: k"; pch = strstr(message, match); if (pch) status++;
+    match = "You put "; pch = strstr(message, match); if (pch) status++;
+    match = "You take "; pch = strstr(message, match); if (pch) status++;
+    match = "You write in the dust with your fingertip"; pch = strstr(message, match); if (pch) status++;
+    match = "Unknown command"; pch = strstr(message, match); if (pch) status++;
+    //match = "What do you want to throw"; pch = strstr(message, match); if (pch) status++;
+    //match = "What do you want to remove"; pch = strstr(message, match); if (pch) status++;
+    //match = "What do you want to put on"; pch = strstr(message, match); if (pch) status++;
+    match = "You were wearing"; pch = strstr(message, match); if (pch) status++;
+    match = "loot it?"; pch = strstr(message, match); if (pch) status++;
+    match = "With great effort you move the boulder"; pch = strstr(message, match); if (pch) status++;
+    match = "For instructions type"; pch = strstr(message, match); if (pch) status++;
+    match = "Check it for traps?"; pch = strstr(message, match); if (pch) status++;
+    match = "not wielded"; pch = strstr(message, match); if (pch) status++;
+    match = "In what direction"; pch = strstr(message, match); if (pch) status++;
+    match = "You find no traps on the"; pch = strstr(message, match); if (pch) status++;
+    match = "Really save?"; pch = strstr(message, match); if (pch) status++;
+    match = "Saving..."; pch = strstr(message, match); if (pch) status++;
+    //match = "What do you want to write with?"; pch = strstr(message, match); if (pch) status++;
+    match = "ynq"; pch = strstr(message, match); if (pch) status++;
+    match = "You wipe out the message that was written in the dust here"; pch = strstr(message, match); if (pch) status++;
+    match = "Something is written here in the dust"; pch = strstr(message, match); if (pch) status++;
+    match = "for sale"; pch = strstr(message, match); if (pch) status++;
+    match = "unpaid"; pch = strstr(message, match); if (pch) status++;
+    match = "Pick an object."; pch = strstr(message, match); if (pch) status++;
+    match = "Sell it?"; pch = strstr(message, match); if (pch) status++;
+    match = "want to name"; pch = strstr(message, match); if (pch) status++;
+    match = "want to call"; pch = strstr(message, match); if (pch) status++;
+    match = "You finish your dressing"; pch = strstr(message, match); if (pch) status++;
+    match = "You finish taking off"; pch = strstr(message, match); if (pch) status++;
+    match = "__ drops"; pch = strstr(message, match); if (pch) status++;
+    match = "__ picks up"; pch = strstr(message, match); if (pch) status++;
+    //match = "What do you want to adjust"; pch = strstr(message, match); if (pch) status++;
+    //match = "What do you want to take off"; pch = strstr(message, match); if (pch) status++;
+    match = "Swapping"; pch = strstr(message, match); if (pch) status++;
+    match = "Choice:"; pch = strstr(message, match); if (pch) status++;
+    match = "Pilgrim"; pch = strstr(message, match); if (pch) status++;
+    match = "intones"; pch = strstr(message, match); if (pch) status++;
+    match = "forbidding feeling"; pch = strstr(message, match); if (pch) status++;
+    match = "You are now wearing"; pch = strstr(message, match); if (pch) status++;
+    match = "Where do you want to travel to"; pch = strstr(message, match); if (pch) status++;
+    match = "The boulder falls into and plugs a hole in the floor"; pch = strstr(message, match); if (pch) status++;
+    match = "p = an"; pch = strstr(message, match); if (pch) status++;
+    match = "Moving"; pch = strstr(message, match); if (pch) status++;
+    match = "You hit the rock with all your might"; pch = strstr(message, match); if (pch) status++;
+    match = "You hit the boulder with all your might"; pch = strstr(message, match); if (pch) status++;
+    match = "Something is engraved here on"; pch = strstr(message, match); if (pch) status++;
+    match = "The thonged club returns"; pch = strstr(message, match); if (pch) status++;
+    match = "The aklys returns"; pch = strstr(message, match); if (pch) status++;
+    match = " - "; pch = strstr(message, match); if (pch) status++;
+    match = " = "; pch = strstr(message, match); if (pch) status++;
+    match = "Nothing happens."; pch = strstr(message, match); if (pch) status++;
+    match = "(I hope you don't mind.)"; pch = strstr(message, match); if (pch) status++;
+    match = "objects here"; pch = strstr(message, match); if (pch) status++;
+    match = "You cannot stash something you are"; pch = strstr(message, match); if (pch) status++;
+    match = "You open the"; pch = strstr(message, match); if (pch) status++;
+    match = "You drop"; pch = strstr(message, match); if (pch) status++;
+    match = "You engrave"; pch = strstr(message, match); if (pch) status++;
+    match = "You see here"; pch = strstr(message, match); if (pch) status++;
+    match = "You make an opening in the wall"; pch = strstr(message, match); if (pch) status++;
+    match = "For you,"; pch = strstr(message, match); if (pch) status++;
+    match = "You hear nothing special"; pch = strstr(message, match); if (pch) status++;
+    match = "Adjust letter to what"; pch = strstr(message, match); if (pch) status++;
+    match = "You read: \"?\"."; pch = strstr(message, match); if (pch) status++;
+    match = "Your armor is rather cumbersome"; pch = strstr(message, match); if (pch) status++;
+    match = "Talk to whom"; pch = strstr(message, match); if (pch) status++;
+    match = "murmurs in your ear"; pch = strstr(message, match); if (pch) status++;
+    match = "Time stands still"; pch = strstr(message, match); if (pch) status++;
+    match = "A boulder fill a pit"; pch = strstr(message, match); if (pch) status++;
+    match = "A boulder fills the pit"; pch = strstr(message, match); if (pch) status++;
+    match = "Shall I pick"; pch = strstr(message, match); if (pch) status++;
+    //match = "Really quit? [yn] (n)"; pch = strstr(message, match); if (pch) status++;
+    match = "You try to move the boulder, but in vain."; pch = strstr(message, match); if (pch) status++;
+    match = "There is nothing here to pick up."; pch = strstr(message, match); if (pch) status++;
+    match = "staircase"; pch = strstr(message, match); if (pch) status++;
+    match = "You fall into the water"; pch = strstr(message, match); if (pch) status++;
+    match = "ock it?"; pch = strstr(message, match); if (pch) status++;
+    match = "Really quit?"; pch = strstr(message, match); if (pch) status++;
+    match = "Unknown type of gloves"; pch = strstr(message, match); if (pch) status++;
+    match = "You try to crawl out of the water."; pch = strstr(message, match); if (pch) status++;
+    match = "Pheew!"; pch = strstr(message, match); if (pch) status++;
+    match = "That was close"; pch = strstr(message, match); if (pch) status++;
+    match = "The Soldier is not ready for battle!"; pch = strstr(message, match); if (pch) status++;
+    match = "sense of peace"; pch = strstr(message, match); if (pch) status++;
+    match = "There is a doorway here."; pch = strstr(message, match); if (pch) status++;
+    match = "There is an open door here."; pch = strstr(message, match); if (pch) status++;
+    match = "Welcome again to"; pch = strstr(message, match); if (pch) status++;
+    match = " asks you for a contribution for the temple."; pch = strstr(message, match); if (pch) status++;
+    match = "Thou art indeed a pious individual."; pch = strstr(message, match); if (pch) status++;
+    match = "You give the priest of "; pch = strstr(message, match); if (pch) status++;
+    match = "Z"; pch = strstr(message, match); if (pch) status++;
+
+    if ( status )
+    {
+    //    status = system(sayit);
+    }
+    else
+    {
+        status = system(sayit);
+        flushinp();
+    }
 }
 
 void
