--- a/src/decl.c
+++ b/src/decl.c
@@ -259,6 +259,25 @@
 };
 
 NEARDATA struct savefile_info sfrestinfo, sfsaveinfo = {
+#ifdef NHSTDC
+    0x00000000UL
+#else
+    0x00000000L
+#endif
+#if defined(COMPRESS) || defined(ZLIB_COMP)
+        | SFI1_EXTERNALCOMP
+#endif
+#if defined(ZEROCOMP)
+        | SFI1_ZEROCOMP
+#endif
+#if defined(RLECOMP)
+        | SFI1_RLECOMP
+#endif
+    ,
+#ifdef NHSTDC
+    0x00000000UL, 0x00000000UL
+#else
+    0x00000000L, 0x00000000L
+#endif
+};
+
+#ifdef VOICE_ENABLED
+NEARDATA char voice_exceptions[MAX_VOICE_EXCEPTIONS][BUFSZ] = { { 0 } };
+NEARDATA int num_voice_exceptions = 0;
+NEARDATA char voice_forces[MAX_VOICE_FORCES][BUFSZ] = { { 0 } };
+NEARDATA int num_voice_forces = 0;
+#endif
 #ifdef NHSTDC
     0x00000000UL
 #else
@@ -277,12 +296,6 @@
 };
 
 struct plinemsg_type *plinemsg_types = (struct plinemsg_type *) 0;
 
 #ifdef VOICE_ENABLED
-NEARDATA char voice_exceptions[MAX_VOICE_EXCEPTIONS][BUFSZ] = { 0 };
-NEARDATA int num_voice_exceptions = 0;
-NEARDATA char voice_forces[MAX_VOICE_FORCES][BUFSZ] = { 0 };
-NEARDATA int num_voice_forces = 0;
-#endif
-#ifdef VOICE_ENABLED
 NEARDATA struct voice_exception *voicelist = (struct voice_exception *) 0; // New linked list for VOICE_EXCEPTION
 NEARDATA struct voice_force *forcelist = (struct voice_force *) 0; // New linked list for VOICE_FORCE
 #endif /* VOICE_ENABLED */
