--- /home/jtratt/s/Release.NetHack-NetHack-3.6.6_Released/sys/unix/sysconf	2020-03-08 12:29:31.000000000 -0500
+++ /home/jtratt/s/voice_Nethack_366/sys/unix/sysconf	2025-01-21 22:52:34.524134644 -0600
@@ -13,7 +13,7 @@
 # Which users can use debug mode (aka wizard mode; accessed via '-D' command
 # line flag or OPTIONS=playmode:debug in the runtime options config file).
 # A value of * allows anyone to enter debugging mode.
-WIZARDS=root games
+WIZARDS=* #root games
 
 # Which users can use explore mode (aka discover mode; accessed via '-X'
 # command line flag or OPTIONS=playmode:explore in runtime options file or
