--- /home/jtratt/s/Release.NetHack-NetHack-3.6.6_Released/src/do_name.c	2020-03-08 12:29:31.000000000 -0500
+++ /home/jtratt/s/voice_Nethack_366/src/do_name.c	2025-02-23 17:34:48.330891266 -0600
@@ -756,8 +756,8 @@
                         dy += ydir[i];
                     }
                 } else {
-                    dx = 8 * xdir[i];
-                    dy = 8 * ydir[i];
+                    dx = 4 * xdir[i];
+                    dy = 4 * ydir[i];
                 }
             } else
                 continue;
