--- /home/jtratt/s/Release.NetHack-NetHack-3.6.6_Released/src/mail.c	2020-03-08 12:29:31.000000000 -0500
+++ /home/jtratt/s/voice_Nethack_366/src/mail.c	2025-02-23 17:34:48.798884790 -0600
@@ -341,7 +341,7 @@
         place_monster(md, fx, fy); /* put md down */
         newsym(fx, fy);            /* see it */
         flush_screen(0);           /* make sure md shows up */
-        delay_output();            /* wait a little bit */
+        nh_delay_output();            /* wait a little bit */
 
         /* Remove md from the dungeon.  Restore original mon, if necessary. */
         remove_monster(fx, fy);
@@ -377,7 +377,7 @@
     place_monster(md, fx, fy); /* place at final spot */
     newsym(fx, fy);
     flush_screen(0);
-    delay_output(); /* wait a little bit */
+    nh_delay_output(); /* wait a little bit */
 
     return TRUE;
 }
