--- /home/jtratt/s/Release.NetHack-NetHack-3.6.6_Released/src/mapglyph.c	2020-03-08 12:29:31.000000000 -0500
+++ /home/jtratt/s/voice_Nethack_366/src/mapglyph.c	2025-02-23 17:34:48.586887724 -0600
@@ -140,9 +140,50 @@
                    && (showsyms[idx] == showsyms[S_pool + SYM_OFF_P]
                        || showsyms[idx] == showsyms[S_water + SYM_OFF_P])) {
             special |= MG_BW_LAVA;
-        } else {
+        //} else {
+        /* Special colors for special dungeon walls */
+        } else if (iflags.use_color && offset >= S_vwall && offset <= S_hcdoor) {
+            if (*in_rooms(x, y, BEEHIVE))
+                color = CLR_YELLOW;
+            else if (In_W_tower(x, y, &u.uz))
+                color = CLR_MAGENTA;
+            else if (In_mines(&u.uz) && !*in_rooms(x, y, 0))
+                color = CLR_BROWN;
+            else if (In_hell(&u.uz) && !Is_valley(&u.uz))
+                color =  CLR_RED;
+            else if (Is_astralevel(&u.uz))
+                color = CLR_WHITE;
+        /* Special colors for special dungeon floors */
+        } else if (iflags.use_color && offset == S_room) {
+            if (*in_rooms(x, y, BEEHIVE))
+                color = CLR_YELLOW;
+            else if (In_hell(&u.uz) && !In_W_tower(x, y, &u.uz))
+                color = Is_juiblex_level(&u.uz) ? CLR_GREEN : CLR_ORANGE;
+        /* Special colors for altars */
+        } else if (iflags.use_color && offset == S_altar) {
+            if (Is_astralevel(&u.uz)) // no cheating!
+                color = CLR_BRIGHT_MAGENTA;
+            else
+                switch((aligntyp)Amask2align(levl[x][y].altarmask & AM_MASK)) {
+                case A_LAWFUL:
+                    color = CLR_WHITE;
+                    break;
+                case A_NEUTRAL:
+                    color = CLR_GRAY;
+                    break;
+                case A_CHAOTIC:
+                    color = CLR_BLACK;
+                    break;
+                default:
+                    color = CLR_RED;
+                    break;
+                }
+       }
+
+       if (color == NO_COLOR)
             cmap_color(offset);
-        }
+        //}
+
     } else if ((offset = (glyph - GLYPH_OBJ_OFF)) >= 0) { /* object */
         idx = objects[offset].oc_class + SYM_OFF_O;
         if (offset == BOULDER)
@@ -241,6 +282,9 @@
         }
     }
 
+    if (iflags.use_color && Is_valley(&u.uz) && color != CLR_BLACK)
+        color = color < NO_COLOR ? CLR_GRAY : CLR_WHITE;
+
     ch = showsyms[idx];
 #ifdef TEXTCOLOR
     /* Turn off color if no color defined, or rogue level w/o PC graphics. */
