--- /home/jtratt/s/Release.NetHack-NetHack-3.6.6_Released/src/pickup.c	2020-03-08 12:29:31.000000000 -0500
+++ /home/jtratt/s/voice_Nethack_366/src/pickup.c	2025-02-23 17:34:48.390890436 -0600
@@ -53,6 +53,9 @@
 #define GOLD_WT(n) (((n) + 50L) / 100L)
 /* if you can figure this out, give yourself a hearty pat on the back... */
 #define GOLD_CAPACITY(w, n) (((w) * -100L) - ((n) + 50L) - 1L)
+#ifdef STICKY_OBJECTS
+extern boolean sticky_inv_hack;        /* blech ick yuck */
+#endif
 
 /* A variable set in use_container(), to be used by the callback routines
    in_container() and out_container() from askchain() and use_container().
@@ -918,11 +921,21 @@
                 }
 
                 any.a_obj = curr;
+#ifdef STICKY_OBJECTS
+           /*
+            * Yes, this is evil and disgusting.  If you have a
+            * better way of doing this, jump on it.
+            */
+           sticky_inv_hack = curr->sticky;
+#endif
                 add_menu(win, obj_to_glyph(curr, rn2_on_display_rng), &any,
                          (qflags & USE_INVLET) ? curr->invlet
                            : (first && curr->oclass == COIN_CLASS) ? '$' : 0,
                          def_oc_syms[(int) objects[curr->otyp].oc_class].sym,
                          ATR_NONE, doname_with_price(curr), MENU_UNSELECTED);
+#ifdef STICKY_OBJECTS
+           sticky_inv_hack = 0;
+#endif
                 first = FALSE;
             }
         }
