--- /home/jtratt/s/Release.NetHack-NetHack-3.6.6_Released/src/shk.c	2020-03-08 12:29:31.000000000 -0500
+++ /home/jtratt/s/voice_Nethack_366/src/shk.c	2025-02-23 17:34:48.734885676 -0600
@@ -3452,7 +3452,7 @@
                   the(xname(obj)));
             if (!canspotmon(shkp))
                 map_invisible(x, y);
-            delay_output();
+            nh_delay_output();
             mark_synch();
         }
         subfrombill(obj, shkp);
